sphinx/source/Installation.rst

```diff
@@ -26,55 +28,90 @@ We officially support building AgIsoStack++ from source on the following platfor
+
 
-In your project that you want to add the CAN stack to, add the CAN stack as a submodule.
+Integration
+-----------
+
+There are multiple ways to integrate the library into your project. 
+The easiest way is to use git submodules in combination with CMake.
+
+Git Submodules
+^^^^^^^^^^^^^^
+
+In directory of the `CMakeLists.txt` that you want to integrate the library into, run the following commands:
 
 .. code-block:: bash
 
    git submodule add https://github.com/Open-Agriculture/AgIsoStack-plus-plus.git
    git submodule update --init --recursive
 
-This will place the CAN stack in a folder within your project called 'AgIsoStack-plus-plus'.
+This will place the stack in a folder within your project called 'AgIsoStack-plus-plus'.
 
-Building the CAN Stack
------------------------
+It is recommended to use the ALIAS targets exposed, which all follow the name `isobus::<target_name>`.
 
-There are a couple options for the next step.
+.. code-block:: cmake
 
-CMake:
-^^^^^^
+   find_package(Threads)
 
-If your project is already using CMake to build your project, or this is a new project, the suggested way to get the library compiling is to add the 'AgIsoStack-plus-plus' folder we just created to your CMake as a subdirectory.
+   add_subdirectory(<path to this submodule>)
 
-.. code-block:: text
+   target_link_libraries(<your executable name> PRIVATE isobus::Isobus isobus::HardwareIntegration isobus::Utility Threads::Threads)
 
-   set(THREADS_PREFER_PTHREAD_FLAG ON)
-   find_package(Threads REQUIRED)
 
-   add_subdirectory("AgIsoStack-plus-plus")
+A full example CMakeLists.txt file can be found at the end of the first tutorial: :doc:`The ISOBUS Hello World <Tutorials/The ISOBUS Hello World>`.
 
-   ...
 
-   target_link_libraries(<your target> PRIVATE isobus::Isobus isobus::HardwareIntegration Threads::Threads)
+Finally, every time you want to update the stack to the latest version, run the following commands:
 
-Using CMake has a lot of advantages, such as if the library is updated with additional files, or the file names change, it will not break your compilation.
+.. code-block:: bash
+
+   git submodule update --remote --merge
+
+This will pull the latest version of the stack into your project.
+
+CMake FetchContent
+^^^^^^^^^^^^^^^^^^
+
+If you don't want to use git submodules, you can use CMake's FetchContent module to download the stack.
+
+.. code-block:: cmake
+
+   include(FetchContent)
+
+   FetchContent_Declare(
+      AgIsoStack
+      GIT_REPOSITORY https://github.com/Open-Agriculture/AgIsoStack-plus-plus.git
+      GIT_TAG        main
+   )
+   FetchContent_MakeAvailable(AgIsoStack)
    
-Non-CMake:
-^^^^^^^^^^
+   # Somewhere later in your CMakeLists.txt
+   target_link_libraries(<your executable name> PRIVATE isobus::Isobus isobus::HardwareIntegration isobus::Utility Threads::Threads)
+
+Now when you configure your CMake cache, the library will be pulled from GitHub and automatically made available for your project.
```

Comment:
We could possibly enhance this section a little to explain that selecting a specific commit may provide better stability and faster configuration times at the expense of not automatically updating 🤔. With a branch selected it has to fetch every time someone configures their CMake cache

Reply:
I'll add that, good thought